ambry-clustermap/src/main/java/com/github/ambry/clustermap/HelixBootstrapUpgradeUtil.java

```diff
@@ -1020,6 +1021,20 @@ private String getResourceTag(String resource) {
    * @param clusterConfigFields disk capacity of each partition.
    */
   private void setClusterConfig(ConfigAccessor configAccessor, ClusterConfigFields clusterConfigFields) {
+    // We will enable TOPOLOGY AWARE in cluster config, we need to make sure all the instances have "DOMAIN" fields
+    HelixConfigScope scope =
+        new HelixConfigScope(HelixConfigScope.ConfigScopeProperty.PARTICIPANT, Collections.singletonList(clusterName),
+            null);
+    // This will list all the participant names
+    List<String> instanceNames = configAccessor.getKeys(scope);
```

Comment:
Hmm, when `--resources` input is given to this tool, I called `setInstanceConfig` only for the instances hosting partitions of those resources (`line number 894`). But, `configAccessor.getKeys(scope)` may get all instances in the cluster. May be we need to set DOMAIN field for all instances in the cluster (irrespective of inputted resources) now ?

Reply:
we should probably do that, let me update it.

Reply:
done